packages/generator/src/generation/generators/table/mui-dt-generator.ts

```diff
@@ -0,0 +1,230 @@
+import GenerationContext from "../../context"
+import { MuiDtTableComponents, muiDataGrid } from '../../../definition/material-ui/table'
+import { TableComponentDefinitionBase } from '../../../definition/table-definition-core'
+import TypescriptHelper from "../../code-generation/ts-helper"
+
+export default class MuiDataTableGenerator extends TableGeneratorBase implements TableGenerator 
+{
+    constructor(generationContext: GenerationContext) {
+        super(generationContext);
+    }
+    
+    generateTableComponent(): PageComponent {
+        var statements = this.createStatements();
+        var functionalComponent = createFunctionalComponent("DataTableComponent", [this.createInputParameter()], statements);
+
+        this._imports = [...this._imports, ...this.intlFormatter.getImports()]
+
+        var uniqueImports = this.uniqueImports();
+        uniqueImports.push(TypescriptHelper.createNameSpaceImport('React', 'react'))
+        
+        return {functionDeclaration: functionalComponent, imports: uniqueImports};
+    }
+
+    getTableDefinition() : TableComponentDefinitionBase {
+        return MuiDtTableComponents;
+    }
+
+    private createStatements(): ts.Statement[] {
+      let statements = new Array<ts.Statement>()
+
+      if(this.context.useFormatter){
+        statements.push(this.intlFormatter.getImperativeHook())
+      }
+
+      let columnsIdentifier = factory.createIdentifier("columns")
+      let columnsDeclaration = this.createColumns(columnsIdentifier)
+
+      var columnsAttribute = createJsxAttribute("columns", "columns")
+      statements.push(factory.createVariableStatement(undefined, columnsDeclaration))
+
+      var rowsAttribute = createJsxAttribute("rows", this.getInputParameterIdentifier())
+
+      let returnStatement = this.createReturnStatement([columnsAttribute, rowsAttribute])
+
+      statements.push(returnStatement)
+
+      return statements;
+    }
+
+    private createReturnStatement(parameters: ts.JsxAttributeLike[]):ts.ReturnStatement {
+      var dataGridComponent = this.prepareComponent(this.getTableDefinition().table);
+
+      let wrappedTable = this.createTableWrapper(createJsxSelfClosingElement(dataGridComponent.tagName, parameters))
+
+      return factory.createReturnStatement(factory.createParenthesizedExpression(wrappedTable))
+    }
+
+    private createTableWrapper(datagrid:ts.JsxSelfClosingElement) {
+      return factory.createJsxElement(
+        factory.createJsxOpeningElement(
+          factory.createIdentifier("div"),
+          undefined,
+          factory.createJsxAttributes([factory.createJsxAttribute(
+            factory.createIdentifier("style"),
+            factory.createJsxExpression(
+              undefined,
+              factory.createObjectLiteralExpression(
+                [
+                  factory.createPropertyAssignment(
+                    factory.createIdentifier("height"),
+                    factory.createNumericLiteral("400")
```

Comment:
should not be hardcoded 
I propose a reasonable default in `context.options.dataTable.height`...